lang/test/org/partiql/lang/eval/NaturalExprValueComparatorsTest.kt

```diff
@@ -364,7 +388,7 @@ class NaturalExprValueComparatorsTest : EvaluatorTestBase() {
     fun nonNullEqualityTests(equivalentPair: Pair<String, String>) {
         val (left, right) = equivalentPair
 
-        assertEval("$left = $right", "true")
+        assertExprEquals(valueFactory.newBoolean(true), eval("$left = $right"))
```

Comment:
Since we have decided to not support `v0` AST serialization for `DATE` and `TIME` nodes, `assertEval` fails because it validates the serialization logic for `DATE` and `TIME`. I have used `assertExprEquals` in its place. What is the right thing to do here?

Reply:
One thing we might want to consider, though I might check with @dlurton, is somehow annotating these cases so we only do `assertExprEquals` in the cases where we don't test the serialization round trips.  That probably looks like changing how the tests are parameterized a bit, but probably the safest way to preserve legacy paths.

Reply:
As discussed, we anyway want to get rid of the round trip tests.

Reply:
To David's point, I think `assertExprEquals` should also suffice instead of asserting separately for value and type. Either way, I don't have any strong inclination towards either one.